src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -36,8 +41,10 @@ public class KustoClient : IKustoClient
         public string ClusterName { get; }
         public string DatabaseName { get; private set; }
 
-        public KustoClient(string connectionString, string azureAccountToken)
+        public KustoClient(string connectionString, string azureAccountToken, string ownerUri)
         {
+            _ownerUri = ownerUri;
+            _retryCount = 1;
```

Comment:
Default to retry once. Saving OwnerUri within the Client so it's able to refresh the token easily.